lib/models/task.js

```diff
@@ -5,7 +5,17 @@ var CoreObject = require('core-object');
 var Task = CoreObject.extend({
   run: function(/*options*/) {
     throw new Error('Task needs to have run() defined.');
-  }
+  },
+
+  /**
+   * Called when the is interrupted from outside, e.g. CTRL+C or process kill
+   * Can be used to cleanup artifacts produced by the task
+   * Return value affects a process exit code
+   *
+   * @method onInterrupt
+   * @return {Promise|undefined} if rejected promise then result of promise will be used as an exit code
+   */
+  onInterrupt: function() {},
```

Comment:
Is this the right default?  I would have thought that for most tasks we would expect a non-zero exit on `SIGINT` (eg npm install, addon install, test, build, generate from blueprint) and the zero-exits on `SIGINT` are the exceptions, for infinite running tasks (basically anything that uses a watcher).